presto-main-base/src/main/java/com/facebook/presto/sql/planner/optimizations/IndexJoinOptimizer.java

```diff
@@ -101,6 +101,10 @@ public PlanOptimizerResult optimize(
         IndexJoinRewriter rewriter;
         if (SystemSessionProperties.isNativeExecutionEnabled(session)) {
             rewriter = new NativeJoinRewriter(idAllocator, metadata, session);
+            if (!rewriter.isPlanChanged()) {
```

Comment:
you're calling this before you try rewriting, so rewriter.isPlanChanged() will always return false.